### PR TITLE
chore(dependencies): remove unused

### PR DESCRIPTION
**Pull request by Haroenv**
- classnames: only used in two little examples and no longer properly declared: removed
- some storybook addons only used in react
- enzyme adaptors only used in react
- glob: used in an old version of the react website
- puppeteer: used in previous tests of react storybook (that didn't run anymore for a while)
- babel plugin that no longer is used
- jscodeshift that isn't used globally
- react-error-boundary that was only used in React InstantSearch (not even sure if it was)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,9 +63,7 @@
     "@testing-library/react": "13.2.0",
     "@testing-library/react-hooks": "7.0.2",
     "@testing-library/user-event": "13.5.0",
-    "@types/classnames": "^2.2.7",
     "@types/enzyme": "^3.1.15",
-    "@types/enzyme-adapter-react-16": "^1.0.3",
```
Comment (Haroenv) [inline package.json:70]: only used by -dom test suite

File: package.json [hunk 2]
```diff
@@ -76,11 +72,8 @@
     "@types/react": "^18.0.9",
     "@types/react-dom": "^18.0.5",
     "@types/react-test-renderer": "^18.0.0",
-    "@types/storybook__addon-a11y": "^5.0.0",
     "@types/storybook__addon-actions": "3.4.2",
     "@types/storybook__addon-knobs": "^5.0.0",
-    "@types/storybook__addon-links": "^3.3.3",
-    "@types/storybook__react": "^4.0.0",
```
Comment (Haroenv) [inline package.json:83]: only used by react storybook

File: package.json [hunk 3]
```diff
@@ -93,12 +86,10 @@
     "@wdio/selenium-standalone-service": "5.16.10",
     "@wdio/spec-reporter": "5.16.11",
     "@wdio/static-server-service": "5.16.10",
-    "@wojtekmaj/enzyme-adapter-react-17": "0.6.5",
     "algoliasearch": "4.14.3",
     "algoliasearch-v3": "npm:algoliasearch@3.35.0",
     "babel-eslint": "10.1.0",
     "babel-jest": "27.4.6",
-    "babel-plugin-dynamic-import-node": "2.3.0",
```
Comment (Haroenv) [inline package.json:101]: only used for react-instantsearch-dom-maps

File: package.json [hunk 4]
```diff
@@ -93,12 +86,10 @@
     "@wdio/selenium-standalone-service": "5.16.10",
     "@wdio/spec-reporter": "5.16.11",
     "@wdio/static-server-service": "5.16.10",
-    "@wojtekmaj/enzyme-adapter-react-17": "0.6.5",
```
Comment (Haroenv) [inline package.json:96]: only used for -dom tests

File: package.json [hunk 5]
```diff
@@ -121,25 +112,21 @@
     "eslint-plugin-react-hooks": "4.3.0",
     "eslint-plugin-vue": "9.8.0",
     "eslint-plugin-wdio": "5.11.0",
-    "glob": "7.1.4",
```
Comment (Haroenv) [inline package.json:124]: used for an old release script of React InstantSearch

File: package.json [hunk 6]
```diff
@@ -121,25 +112,21 @@
...
     "jest": "27.4.7",
     "jest-diff": "27.4.6",
     "jest-environment-jsdom": "27.1.0",
     "jest-environment-jsdom-global": "2.0.4",
     "jest-junit": "12.2.0",
     "jest-vue-preprocessor": "1.7.1",
     "jest-watch-typeahead": "1.0.0",
-    "jscodeshift": "0.7.0",
```
Comment (Haroenv) [inline package.json:132]: duplicate with the codemods now (used by InstantSearch for testing before)

File: package.json [hunk 7]
```diff
@@ -121,25 +112,21 @@
...
     "jsdom-global": "3.0.2",
     "lerna": "6.0.3",
     "patch-package": "6.2.2",
     "places.js": "1.17.1",
     "prettier": "2.4.1",
     "prop-types": "15.6.2",
-    "puppeteer": "1.20.0",
```
Comment (Haroenv) [inline package.json:139]: used for old tests on react storybook (that didn't run anymore)

File: package.json [hunk 8]
```diff
@@ -121,25 +112,21 @@
...
     "react": "18.1.0",
     "react-dom": "18.1.0",
-    "react-error-boundary": "3.1.3",
```
Comment (Haroenv) [inline package.json:142]: was only used in the react-core dynamic widgets test

**Review by dhayab — APPROVED** [review]
Great cleanup, thanks!

**Pull request merged**